small simple bug-fix: update lost infos in function create_dataset function

im-Kitsch opened the pull request:
# Description

This is fix for simple bug-fix for function `create_dataset_from_collector_env()`.  
1. Add docstring for parameters
2. Add metadata attribute to dataset, meta data like `algorithm name` `author` `author Email` is not saved though it's passed into function.

for function `create_dataset_from_collector_env()`
ref_min/ref_max score is not saved to metadata, it's easily to find the problem.

I think the bugs are quite clear and definitely is a bug, you can take a look,

## Type of change

- Bug fix (non-breaking change which fixes an issue)


# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

<!--
You can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

balisujohn commented:
Good catches. Thanks for your contribution! All these fixes look correct to me.  I think it would be good to make the values for `author_name`, `author_email`, `code_permalink`  default to `"not provided"` instead of `str(None)` if not provided. Once that's done I will merge this PR after a final review.  As for minari version as an argument, I'm wondering if we should make it non-optional, but we can sort that out in a future PR.

im-Kitsch commented:
Hi, 

I pushed update to save `author_name`, `author_email`, `code_permalink` and `algorithm name` only when it's not None.  And additionally I changed a code of combine dataset so that it could optional parameters are not saved. 

I agree that we need consider if minari version should be optional or mandatory. 

Futhermore, I think the dataset's structure should be **clearly** specified and let it be **stable**. 

E.g. here in `combine_dataset`, if `author` or `author_email` are not saved then it will throw error. 
https://github.com/Farama-Foundation/Minari/blob/ea4097854cae1b333ddd7de2fc3fabd0fbc1076d/minari/utils.py#L277-L290

There should be still small errors like this one.

balisujohn commented on minari/utils.py:
```diff
@@ -284,10 +284,15 @@ def combine_datasets(
+                        combined_data_file.attrs.modify(
+                            optional_parameter, dataset_file.attrs[optional_parameter]
+                        )
+                # combined_data_file.attrs.modify("author", dataset_file.attrs["author"])
```
please remove commented out lines

balisujohn reviewed (changes requested):
The code seems almost ready to merge. Please remove the commented out lines, and add a basic test to for behavior when the  optional `dataset_metadata` keys modified in this PR are provided when creating an instance of `MinariDataset`

im-Kitsch commented:
Hi, 
I added one test case for `combine dataset` and modified `create dataset` test case, could you take a look? 

I think these tests should be modified in the future again, since currently those additional infomations could not be got directly from `Minaridataset` instance. But have to be read manualy from h5py file.

younik commented on minari/utils.py:
```diff
@@ -486,6 +503,19 @@ def create_dataset_from_buffers(
                 "env_spec"
             ] = env.spec.to_json()  # pyright: ignore [reportOptionalMemberAccess]
             file.attrs["dataset_id"] = dataset_id
+            # file.attrs["algorithm_name"] = str(algorithm_name)
+            # file.attrs["author"] = str(author)
+            # file.attrs["author_email"] = str(author_email)
+            # file.attrs["code_permalink"] = str(code_permalink)
```
can you remove these comments?

younik commented on tests/utils/test_dataset_combine.py:
```diff
@@ -231,3 +232,53 @@ def test_combine_minari_version_specifiers(specifier_intersection, version_speci
+            dataset_id=dataset_id,
+            collector_env=env,
+            algorithm_name="random_policy" + str(i),
+            code_permalink="https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
```
can you change the permalink to be the following? https://github.com/Farama-Foundation/Minari/blob/main/tests/utils/test_dataset_combine.py

younik commented on tests/utils/test_dataset_combine.py:
```diff
@@ -231,3 +232,53 @@ def test_combine_minari_version_specifiers(specifier_intersection, version_speci
+    with h5py.File(combined_dataset.spec.data_path) as dt_file:
+        assert dt_file.attrs["algorithm_name"] == "random_policy" + str(n_data - 1)
+        _final_code_link = (
+            "https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
+            + str(n_data - 1)
```
same here

younik commented on tests/utils/test_dataset_creation.py:
```diff
@@ -72,6 +73,14 @@ def test_generate_dataset_with_collector_env(dataset_id, env_id):
+    # test metadata
+    with h5py.File(dataset.spec.data_path, "r") as data_file:
+        assert data_file.attrs["algorithm_name"] == "random_policy"
+        code_link = "https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
```
same here

younik commented on tests/utils/test_dataset_creation.py:
```diff
@@ -279,6 +288,14 @@ def test_generate_dataset_with_space_subset_external_buffer():
+    # test metadata
+    with h5py.File(dataset.spec.data_path, "r") as data_file:
+        assert data_file.attrs["algorithm_name"] == "random_policy"
+        code_link = "https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
```
same ehre

younik commented on tests/utils/test_dataset_combine.py:
```diff
@@ -231,3 +232,53 @@ def test_combine_minari_version_specifiers(specifier_intersection, version_speci
+            "https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
+            + str(n_data - 1)
+        )
+        assert dt_file.attrs["code_permalink"] == _final_code_link
+        assert dt_file.attrs["author"] == "WillDudley" + str(n_data - 1)
+        assert dt_file.attrs["author_email"] == "wdudley@farama.org" + str(n_data - 1)
```
btw, this behavior is a little weird in combine dataset (you just save the attributes of the last)
Anyway, it is fixed in https://github.com/Farama-Foundation/Minari/pull/133 that we will merge after, so it is okay for now

im-Kitsch replied:
Ah, yes, I don't think only saving the attribute of last is a good choice, but it's just to keep same with `combine_dataset()` implementation. Definitely it will be changed in the future, but this PR is just to save the fogortten metadata `code_permalink` and `algorithm_name`.

younik reviewed (approved):
Just fix the pre-commit, and then looks good to me

im-Kitsch commented:
> Just fix the pre-commit, and then looks good to me

Hi, I think the pre-commit is fixed now.
